mayastor/src/bdev/dev/nvmx/controller.rs

```diff
@@ -502,6 +529,38 @@ impl<'a> NvmeController<'a> {
         );
         NvmeController::_complete_reset(reset_ctx, status);
     }
+
+    fn notify_event(&self, event: DeviceEventType) -> usize {
+        // Keep a separate copy of all registered listeners in order to not
+        // invoke them with the lock held.
+        let listeners = {
+            let listeners = self
+                .event_listeners
+                .lock()
+                .expect("event listeners lock poisoned");
+            listeners.clone()
+        };
+
+        for l in listeners.iter() {
+            (*l)(event, &self.name);
+        }
+        listeners.len()
+    }
+
+    /// Register listener to monitor device events related to this controller.
+    pub fn add_event_listener(
+        &self,
+        listener: fn(DeviceEventType, &str),
+    ) -> Result<(), CoreError> {
```

Comment:
nit; we never generate an error here.

Reply:
I wanted to keep the API more flexible. Later on we might add checks against the maximum limit of event listeners, etc.